### PR TITLE
Clickable Cards

### PR DESCRIPTION
**Pull request by cyberpwnn**
Simply allows cards to be pressed with an onPressed VoidCallback. For compatibility it simply uses a GhostButton in the cards child. In the event that a card is clickable, it's padding is used in the ghost button and not the card, otherwise it functions as cards always have.

### CODE REVIEW

**Comment by NextdoorPsycho** [comment]
I Would use this!

**Comment by sunarya-thito** [comment]
While the idea is great, I think this should be done with another approach. I believe the card, being part of the layout component, should not be directly clickable. I have pushed several commits to implement this feature with another approach. Let me know what else is needed or if you think this pull request is still necessary. Next time please open an issue first so we can discuss the possible approach when adding new component capability.

**Comment by cyberpwnn** [comment]
Some designs with a grid of cards in some apps especially content management apps have loads of different data models that all need this kind of layout, making something clickable vs not means conditional builds, it's significantly easier to just add / nullify an onPressed property. 

Only other alternative would be creating a CardButton widget, mirroring all of the card properties which it looks like you have, if you are maintaining that debt of keeping property parity between CardButton and Card then it's fine, but I'm mostly trying to reduce the amount of breaking changes I introduce to myself across all my projects, which is why I diddnt create my own CardButton widget and instead just opted to add the property to Card.

I use my fork of shadcn, not this one, this change has been in there for a while and I've found it really nice to add functionality to existing widgets so If you feel this isnt really a good fit, that's fine as I'm essentially rebasing your changes into my repo for anything that doesn't fit into shadcn_flutter nicely. 

In my mind its just an optional field, while it doesn't "fit" in the layout/surface/control model, I'm more in the business of getting functional & reliable UI out the door without having juggle widgets on my clipboard, or refactor lots of files. I believe this change is a great example that.

TLDR: Would have to refactor files to use CardButton, If its no big deal to you, merging means less conflict resolution my fork has to do to stay updated, not merging is also fine because my products use my fork of shadcn_flutter and rebasing is always an option for me.

**Comment by cyberpwnn** [comment]
I'll just use the CardButton then
